File: sang-math-omr/check_tf.py
import re
import json

with open("index.html", "r") as f:
    html = f.read()

match = re.search(r"'12-4-6ngang':\s*(\{.*?\})\s*,\s*'tn-50'", html, re.DOTALL)
if match:
    # We can't eval js to json easily, let's just regex find "tf": { ... }
    tf_match = re.search(r'"tf":\s*(\{.*?\})', match.group(1), re.DOTALL)
    if tf_match:
        print(tf_match.group(0)[:500])
    else:
        print("tf not found in 12-4-6ngang")
else:
    print("12-4-6ngang not found")

File: sang-math-omr/check_tf.js
const fs = require('fs');
const html = fs.readFileSync('index.html', 'utf8');
const match = html.match(/'12-4-6ngang': (\{[\s\S]*?\})\s*,\s*'tn-50'/);
if (match) {
    const tmplStr = match[1];
    // console.log(tmplStr.substring(0, 500));
    const tmpl = eval('(' + tmplStr + ')');
    console.log(JSON.stringify(tmpl.tf, null, 2));
} else {
    console.log('not found');
}
